stream/src/main/java/com/instaclustr/kafka/connect/stream/endpoint/S3BucketAws.java

```diff
@@ -0,0 +1,51 @@
+package com.instaclustr.kafka.connect.stream.endpoint;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+import com.instaclustr.kafka.connect.stream.Endpoint;
+import org.apache.kafka.common.config.AbstractConfig;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+public class S3BucketAws extends S3Bucket implements AccessKeyBased {
+
+    // Visible for testing
+    S3BucketAws(TransferManager transferManager, String bucketName) {
+        super(transferManager, bucketName);
+    }
+
+    public static S3BucketAws of(Map<String, String> providedConf) {
+        AbstractConfig s3BucketConf = new AbstractConfig(S3Bucket.CONFIG_DEF, providedConf);
+        AbstractConfig accessKeyConf = new AbstractConfig(AccessKeyBased.CONFIG_DEF, providedConf);
+
+        AWSStaticCredentialsProvider awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(accessKeyConf.getString(ACCESS_KEY_ID), accessKeyConf.getPassword(ACCESS_KEY).value())
+        );
+
+        AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder.standard()
```

Comment:
In our S3 connector we also allow specifying a IAM role ARN. Would be nice to have that capability for this as well. But happy for it to be a future feature - with a medium ticket